module/vision/FieldLineDetector/src/FieldLineDetector.cpp

```diff
@@ -0,0 +1,96 @@
+#include "FieldLineDetector.hpp"
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <fmt/format.h>
+#include <numeric>
+
+#include "extension/Configuration.hpp"
+
+#include "message/support/FieldDescription.hpp"
+#include "message/vision/FieldLines.hpp"
+#include "message/vision/GreenHorizon.hpp"
+
+#include "utility/math/coordinates.hpp"
+#include "utility/support/yaml_expression.hpp"
+#include "utility/vision/visualmesh/VisualMesh.hpp"
+
+
+namespace module::vision {
+
+    using extension::Configuration;
+
+    using message::support::FieldDescription;
+    using message::vision::FieldLines;
+    using message::vision::GreenHorizon;
+
+    FieldLineDetector::FieldLineDetector(std::unique_ptr<NUClear::Environment> environment)
+        : Reactor(std::move(environment)), config{} {
+
+        on<Configuration>("FieldLineDetector.yaml").then([this](const Configuration& cfg) {
+            // Use configuration here from file FieldLineDetector.yaml
+            this->log_level = cfg["log_level"].as<NUClear::LogLevel>();
+
+            this->config.confidence_threshold = cfg["confidence_threshold"].as<float>();
+            this->config.cluster_points       = cfg["cluster_points"].as<int>();
+        });
+
+        on<Trigger<GreenHorizon>, Buffer<2>>().then("Field Line Detector", [this](const GreenHorizon& horizon) {
```

Comment:
how come we have the buffer?

Reply:
Not exactly sure, I imagine this module was first just duplicate of BallDetector which has the buffer. Goal detector has it as well.

Also in regards to this module, (Note: I haven't delved into what `partition_points` and `cluster_points` exactly do), from reading the comments their may be an easier way to extract field line points from mesh, since clustering isn't required here.